src/deepsparse/pipeline.py

```diff
@@ -700,15 +704,24 @@ def to_config(self) -> "PipelineConfig":
             kwargs=kwargs,
         )
 
-    def log(self, identifier: str, value: Any, category: str):
+    def log(
+        self,
+        identifier: str,
+        value: Any,
+        category: str,
+        identifier_pre_fix: Optional[str] = None,
```

Comment:
This would probably not solve our problem.
Let me illustrate why, by giving a short example.

Let's say that we are logging the prediction latency ( of the `pre_process` step) of a pipeline called `sample_pipeline`.

Right now, with the simplified design of identifiers and "identifier pre-fix", the pipeline would emit an identifier: `prediction_latency/sample_pipeline/pre_process`. The function logger that is "looking for" this identifier would use the target identifier `prediction_latency` and simply check for the match between the first components to the target identifier and identifier strings.

Without the identifier pre-fix, we are losing this neat matching functionality. If the pipeline were to emit an identifier `sample_pipeline/prediction_latency/pre_process` (this would happen if I followed your advice), matching would get quite complex. Especially in the situation where our pipeline becomes complex/nested. 

We can try to make it work, but it would significantly complicate the matching logic. And I think the matching logic is something that we want to keep extremely simple, otherwise, it may get unnecessarily complex quickly.

But I am open to discussing this further.